### ISSUE TITLE
taskwarrior build failure and build dependencies in general

### ISSUE THREAD

**Issue by berryamin** [BerryFarm/berrymuch#26]
@yamsergey I saw somewhere that taskwarrior build failed in the CI.
As a simple fix for now, here is what I would suggest:
- take advantage of the following feature inside lib.sh:

lib.sh: bootstrap () : skip build of package when packages/$DISTVER.zip is already present and print a Warning (5c58ff9e9b81d5d9809b5fc554c58187fb096d7a)

taskwarrior/build.sh : put a "../../ports/<required_library>/build.sh".

But here comes a question .. Is there a persistence of the work/ directory accross builds or is it re-created each time ? 

If there is persistence of work/ , removing package/<required_library-version>.zip would be enough to fix this issue.

If there is no persistence of work/ .. a mecanism to "unpack" zip from packages/<required_library-version> has to be used/designed.

What do you think :) ?

**Comment by yamsergey** [comment]
No, docket container doesn’t have state from previous builds, new one spin up each time. I want to dig inside Termux approach to build.

Tanks for the insight about lib.sh. I will think how can we utilize it during build

**Comment by berryamin** [comment]
Oups .. the "skip port if already built" feature is based on "vars.sh" ...

<pre>

  for dir in $ALLPORTS
  do
    if [ -d "$dir" ] && [ -e "$dir/build.sh" ]; then
      [ -f "$dir/vars.sh" ] && {
        . $dir/vars.sh
        [ -f ../packages/$DISTVER.zip ] && {
          echo "Warning: skipping build of $DISTVER (file packages/$DISTVER.zip already exists) "
          continue;
        }
      }

</pre>

vars.sh was removed from everywhere a short while ago ...

**Comment by vaskas** [comment]
We are currently tracking dependencies by hand putting libraries in order but this will get messy quickly

**Comment by berryamin** [comment]
For gdb (needs gmp), I had to write this little trick, currently only available on my local "bleeding-edge" branch:

lib.sh:

<pre>
function unpack_required_packages()
{
        I=0
        while [ -z != ${BUILD_DEP[$I]} ];
        do
                pushd $WORKDIR > /dev/null || exit 1
                        unzip /$PKGDIR/${BUILD_DEP[$I]}.zip
                popd > /dev/null
                let I=$(($I+1))
        done
}

function check_required_packages()
{
        I=0
        while [ -z != ${BUILD_DEP[$I]} ];
        do
                F=$PKGDIR/${BUILD_DEP[$I]}.zip
                ls $F || {
                        echo "Fatal: package ${BUILD_DEP[$I]} not found. Required for build"
                        exit 1
                }
                let I=$(($I+1))
        done
}


</pre>

**Comment by vaskas** [comment]
Hmm I think you shouldn't need this unless you are wiping archive/

**Comment by berryamin** [comment]
yes indeed. But Sergey told me there is no persistence on the build farm :)
